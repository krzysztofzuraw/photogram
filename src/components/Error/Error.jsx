import React from "react";
import PropTypes from "prop-types";

const Error = ({ message, onClick }) => {
  return (
    <div>
      <span>There was some error with fetching data from the server.</span>
      <span>Error message: {message}</span>
      <button>Retry?</button>
    </div>
  );
};

Error.PropTypes = {
  message: PropTypes.string,
  onClick: PropTypes.func
};

export default Error;
