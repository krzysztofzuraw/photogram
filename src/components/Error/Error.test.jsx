import React from "react";

import { shallow } from "enzyme";

import Error from "./Error";

describe("Error Component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Error />);
    expect(wrapper).toMatchSnapshot();
  });
});
