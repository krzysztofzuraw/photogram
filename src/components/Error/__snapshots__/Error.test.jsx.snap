// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Error />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          There was some error with fetching data from the server.
</span>,
        <span>
          Error message: 
</span>,
        <button>
          Retry?
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "There was some error with fetching data from the server.",
        },
        "ref": null,
        "rendered": "There was some error with fetching data from the server.",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Error message: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Error message: ",
          undefined,
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Retry?",
        },
        "ref": null,
        "rendered": "Retry?",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            There was some error with fetching data from the server.
</span>,
          <span>
            Error message: 
</span>,
          <button>
            Retry?
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "There was some error with fetching data from the server.",
          },
          "ref": null,
          "rendered": "There was some error with fetching data from the server.",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Error message: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Error message: ",
            undefined,
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Retry?",
          },
          "ref": null,
          "rendered": "Retry?",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
